packages/app-tests/e2e/test-document-flow.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { expect } from '@playwright/test';
+
+import { test } from '../fixtures/fixtures';
+
+test.use({ storageState: { cookies: [], origins: [] } });
+
+const testSigners = [
+  { email: 'signer1@gmail.com', name: 'signer 1' },
+  { email: 'signer2@gmail.com', name: 'signer 2' },
+  { email: 'signer3@gmail.com', name: 'signer 3' },
+];
+
+test.describe('test document flow', async () => {
+  test.beforeEach(async ({ users }) => {
+    const user = await users.create();
+    await user.apiLogin();
+  });
+  test.afterEach(async ({ users }) => users.deleteAll());
+
+  test('user can upload documents', async ({ page, samplePdf }) => {
+    await page.goto('/documents');
+    await expect(page).toHaveURL('/documents');
+
+    await page.getByTestId('document-dropzone').setInputFiles({
+      buffer: samplePdf,
+      mimeType: 'application/pdf',
+      name: 'test.pdf',
+    });
+
+    await expect(page).toHaveURL(/\/documents\/\d+/);
+  });
+
+  test('user can add signers', async ({ page, samplePdf }) => {
+    await page.goto('/documents');
+    await expect(page).toHaveURL('/documents');
+
+    await page.getByTestId('document-dropzone').setInputFiles({
+      buffer: samplePdf,
+      mimeType: 'application/pdf',
+      name: 'test.pdf',
+    });
+
+    await expect(page).toHaveURL(/\/documents\/\d+/);
```

Comment:
lines 34-43 are repeated in 2 tests. I suppose they'll repeat again if we add another test for this page. Can we re-use them somehow or re-structure the tests?

Reply:
now refactored